plugins/woocommerce/includes/data-stores/class-wc-customer-download-data-store.php

```diff
@@ -240,11 +240,20 @@ public function update( &$download ) {
 	public function delete( &$download, $args = array() ) {
 		global $wpdb;
 
+		$download_id = $download->get_id();
 		$wpdb->query(
 			$wpdb->prepare(
 				"DELETE FROM {$wpdb->prefix}woocommerce_downloadable_product_permissions
 				WHERE permission_id = %d",
-				$download->get_id()
+				$download_id
+			)
+		);
+		// Delete related records in wc_download_log (aka ON DELETE CASCADE).
+		$wpdb->query(
+			$wpdb->prepare(
+				"DELETE FROM {$wpdb->prefix}wc_download_log
+				WHERE permission_id = %d",
+				$download_id
 			)
 		);
```

Comment:
This code is repeated verbatim shortly after. Could it be refactored into a `delete_download_log_by_permission_id` method (private and in the same class)?

Reply:
Good point. I think I can just call `$this->delete_by_id( $download_id )` and that will take care of both queries.

Reply:
Should be fixed in[461d5c4](https://github.com/woocommerce/woocommerce/pull/34318/commits/461d5c4bb6d50cf752242b00a4d0ca70b5dcc227).